app/Http/Controllers/loginController.php

```diff
@@ -0,0 +1,176 @@
+                'status'           => '401',
+                'error'            => $validator->messages(),
+                'validation_error' => 1,
+            ]);
+        }
+
+        $user = User::where('email', $req->email)->get()->first();
+        if ($user) {
+            $emailExists = Resetpassword::where('email', $req->email)->get()->first();
+            $otp = random_int(100000, 999999);
+            if ($emailExists) {
+                $emailExists->otp = $otp;
+                $emailExists->save();
+                $id = $emailExists->id;
+            } else {
+                $row = new ResetPassword();
+                $row->email = $req->email;
+                $row->otp = $otp;
+                $row->save();
+                $id = $row->id;
+            }
+            $resetPasswordLink = url('checkLink'.'/'.$id.'/'.$otp);
+            $emailData = ['link' => $resetPasswordLink, 'name' => $user->name];
+            Mail::to($req->email)->send(new resetpasswordemail($emailData));
+
+            return response()->json([
+                'success'=> trans('lang.Success_Link_Intro'),
+                'status' => 200,
+            ]);
+        } else {
+            return response()->json([
+                'error'            => trans('lang.Invalid_Email'),
+                'status'           => 401,
+                'validation_error' => 0,
+            ]);
+        }
+    }
+
+    public function checkOtp($id, $otp)
+    {
+        $user = Resetpassword::find($id);
+        if ($user) {
+            if ($user->otp == $otp) {
+                Cache::add('email', $user->email);
+
+                return redirect('setpassword');
+            }
+
+            return redirect('/')->with('error', trans('lang.Invalid_OTP'));
+        }
+
+        return redirect('/')->with('error', trans('lang.Invalid_Email'));
+    }
+
+    public function setPassword(Request $req)
+    {
+        $validator = Validator::make($req->all(), [
+            'password'          => ['required', Password::min(6)->letters()->numbers()->symbols()],
+            'confirmpassword'   => ['required'],
+        ]);
+        if ($validator->fails()) {
+            return response()->json([
+                'status'           => '401',
+                'error'            => $validator->messages(),
+                'validation_error' => 1,
+            ]);
+        }
+        $user = User::where('email', Cache::get('email'))->get()->first();
+        if ($user) {
+            if ($req->password == $req->confirmpassword) {
+                $id = $user->id;
+                $data = User::find($id);
+                $data->password = Hash::make($req->password);
+                $data->save();
+                Cache::forget('email');
+
+                return response()->json([
+                    'success'=> trans('lang.Success_Password_Intro'),
+                    'status' => 200,
+                ]);
+            }
+
+            return response()->json([
```

Comment:
create response JSON method in helper file, like success response nad error response. Take a reference from faveo